packages/typespec-test/test/openai_modular/generated/typespec-ts/src/utils/serializeUtil.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  ChatRequestUserMessage as ChatRequestUserMessageRest,
+  ChatRequestAssistantMessage as ChatRequestAssistantMessageRest,
+  ChatRequestToolMessage as ChatRequestToolMessageRest,
+  ChatRequestMessage as ChatRequestMessageRest,
+  ChatMessageImageContentItem as ChatMessageImageContentItemRest,
+  ChatMessageContentItem as ChatMessageContentItemRest,
+} from "../rest/index.js";
+import {
+  ChatRequestUserMessage,
+  ChatRequestAssistantMessage,
+  ChatRequestToolMessage,
+  ChatRequestMessage,
+  ChatMessageImageContentItem,
+  ChatMessageContentItem,
+} from "../models/models.js";
+
+/** serialize function for ChatRequestUserMessage */
+function serializeChatRequestUserMessage(
+  obj: ChatRequestUserMessage
+): ChatRequestUserMessageRest {
+  return {
+    role: obj["role"],
+    content: obj["content"] as any,
```

Comment:
In this case, the type of content is defined as `alias ChatMessageContent = string | ChatMessageContentItem[];`  and `ChatMessageContentItem` is another polymohic base with special subtype. which we could handle here https://github.com/Azure/autorest.typescript/pull/2169/files#diff-1a35b9806b61290726e7a05f336d1849f89253f8e4c1591a991919a90b578733R90-R100, but because of the extra variant string, we just generate it as any now. Not sure if you have other thoughts here ? @joheredi @xirzec @MaryGao